examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,432 @@
+# for each choice of
+# gate :math:`R_d \in \{R_x,R_y,R_z\}` and selecting the gate which yields the minimum value of
+# :math:`\langle H \rangle`.
+#
+# Thus, one might expect the number of circuit evaluations required to be 9 for each parameter (3 for each gate
+# choice). However, since there is a 3-fold
+# degeneracy in the expectation value when :math:`\theta_d=0`,
+#
+# .. math:: R_x(0) = R_y(0) = R_z(0) = 1,
+#
+# the number of evaluations is reduced to 7.
+#
+# One cycle of the Rotosolve algorithm involves
+# iterating through every parameter and performing the calculation above.
+# This cycle is repeated for a fixed number of steps or until convergence. In this way, one could learn both
+# the optimal parameters and gate choices for a circuit ansatz. Next, we present an example of this algorithm
+# applied to a VQE Hamiltonian.
+#
+# Example VQE Problem
+# -------------------
+#
+# We focus on a 2-qubit VQE circuit for simplicity. Here, the Hamiltonian
+# is
+#
+# .. math::
+#   H = 0.5Y_2 - 0.8Z_1 - 0.2X_1
+#
+# where the subscript denotes the qubit upon which the Pauli operator acts. The
+# expectation value of this quantity acts as the cost function for our
+# optimization.
+#
+# Rotosolve
+# ---------
+# As a precursor to implementing Rotoselect we can analyze a version of the algorithm
+# which does not optimize the choice of gates, called Rotosolve. Later, we will build on this example
+# to implement Rotoselect and vary the circuit structure.
+#
+# Imports
+# ~~~~~~~
+# To get started, we import PennyLane and the PennyLane-included version of NumPy. We also
+# create a 2-qubit device using the ``default.qubit`` plugin and set the ``analytic`` keyword to ``True``
+# in order to obtain exact values for any expectation values calculated. In contrast with real
+# devices, simulators have the capability of doing these calculations without sampling.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+n_wires = 2
+
+dev = qml.device("default.qubit", analytic=True, wires=2)
+
+##############################################################################
+# Creating a fixed quantum circuit
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# .. figure:: ../../examples/figures/original_ansatz.png
+#    :scale: 65%
+#    :align: center
+#    :alt: original_ansatz
+#
+# |
+# Next, we set up a circuit with a fixed ansatz structure---which will later be subject to change---and encode
+# the Hamiltonian into a cost function. The structure is shown in the figure above.
+
+
+def ansatz(params):
+    qml.RX(params[0], wires=0)
+    qml.RY(params[1], wires=1)
+    qml.CNOT(wires=[0, 1])
+
+
+@qml.qnode(dev)
+def circuit(params):
+    ansatz(params)
+    return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))
+
+
+@qml.qnode(dev)
+def circuit2(params):
+    ansatz(params)
+    return qml.expval(qml.PauliX(0))
+
+
+def cost(params):
+    Z_1, Y_2 = circuit(params)
+    X_1 = circuit2(params)
+    return 0.5 * Y_2 + 0.8 * Z_1 - 0.2 * X_1
```

Comment:
The sign of Z_1 does not match the Hamiltonian above

Reply:
Thank you!